src/Container.js

```diff
@@ -0,0 +1,431 @@
+        'Content-Type': 'application/json'
+      }
+    }).then(response => {
+      if (200 !== response.status) {
+        return;
+      }
+      response.json().then(json => {
+        const release = json.data;
+        const error = Features.test(release.capabilities);
+        if (error) {
+          return CLIENT.trigger('unsupported');
+        }
+
+        this.release = release;
+        this._internalOpen(release.url + query, { singlePlay, playOptions });
+      });
+    });
+  }
+
+  /**
+   * Destroy and don't use after this
+   * @memberof Container
+   */
+  destroy() {
+    this.reset();
+    // Destroy in the reverse priority order
+    this.plugins
+      .slice()
+      .reverse()
+      .forEach(plugin => plugin.teardown(this));
+
+    this.main = null;
+    this.options = null;
+    this.dom = null;
+    this.release = null;
+  }
+
+  /**
+   * Tell the application to start closing
+   * @memberof Container
+   */
+  close() {
+    if (this.loading || this.loaded) {
+      this.plugins.forEach(plugin => plugin.close(this));
+      CLIENT.trigger('close');
+      // Start the close
+      CLIENT.send('close');
+    } else {
+      this.reset();
+    }
+  }
+
+  /**
+   * Runs all plugin require preload functions
+   * @async
+   * @memberof Container
+   */
+  preload() {
+    const preloader = Promise.resolve();
+    this.plugins.forEach(plugin => preloader.then(() => plugin.preload()));
+    preloader
+      .then(() => CLIENT.send('plugins loaded'))
+      .catch(e =>
+        console.error('SpringRoll Container Plugin Preloader error: ', e)
+      );
+  }
+
+  /**
+   * The current version of SpringRollContainer
+   * @readonly
+   * @static
+   * @memberof Container
+   */
+  static get version() {
+    return version;
+  }
+
+  /**
+   * On Container instantiation all plugins based through this function will be supplied to the Container
+   * @static
+   * @param {SpringRollContainer.BasePlugin} plugin your plugin. This will be merged with a base plugin to make sure your plugin has certain functions
+   * @memberof Container
+   */
+  static uses(plugin) {
+    PLUGINS.push(plugin);
+  }
+
+  /**
+   * @readonly
+   * @static
+   * @memberof Container
+   * @returns {Array<SpringRollContainer.BasePlugin>}
+   */
+  static get plugins() {
+    return PLUGINS;
+  }
+
+  /**
+   * Clears all plugin instances from Container
+   * @static
+   * @memberof Container
+   */
+  static clearPlugins() {
+    PLUGINS = [];
+    return PLUGINS;
+  }
+
+  /**
+   * @readonly
+   * @static
+   * @returns {Bellhop}
+   * @memberof Container
+   */
+  static get client() {
+    return CLIENT;
+  }
+
+  /**
+   * @readonly
+   * @returns {Bellhop}
+   * @memberof Container
+   */
+  get client() {
+    return CLIENT;
+  }
+
+  /**
+   * Helper method for sorting plugins in place. Looks at dependency order and performs a topological sort to enforce
+   * proper load error
+   */
+  static sortPlugins() {
+    if (PLUGINS.length === 0) {
```

Comment:
Do we want to convert this over to the newer plugin ordering mechanism we used in SR2? In short, we [added methods](https://github.com/SpringRoll/SpringRoll/tree/v2/src/plugins#execution-order-and-intent) (`init` and `start`) to allow plugins to depend on each other. It's probably best that we synchronize these behaviors for plugin authors.

Reply:
For the sake of not duplicating code, would it be a good idea to move all the plugin code into its own library that both Core and Container can share?